src/App.tsx

```diff
@@ -15,47 +15,114 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type IsReserved = boolean;
+
 export const App: React.FC = () => {
+  const [isReversed, setReserved] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE as SortType);
+
+  const getReorderedGoods = (
```

Comment:
you should move this function top, outside of App component to avoid re rendering every time